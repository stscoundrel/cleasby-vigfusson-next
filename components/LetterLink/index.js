import Link from 'next/link'

export default function LetterLink({ letter }) {
  return (
   <Link href={`/letter/${letter}`}>
      <a>{letter}</a>
    </Link>
  )
}
