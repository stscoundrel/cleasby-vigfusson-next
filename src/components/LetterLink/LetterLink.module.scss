.link {
  color: white;
  text-decoration: none;
  padding: 5px 10px;

  &:hover {
    text-decoration: underline;
  }
}