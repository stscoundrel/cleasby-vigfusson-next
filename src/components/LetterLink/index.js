import Link from 'next/link'
import styles from './LetterLink.module.scss'

export default function LetterLink({ letter }) {
  return (
    <Link
      href={`/letter/${letter.slug}`}
      className={styles.link}
      prefetch={false}
    >
       {letter.letter}
     </Link>
  );
}
