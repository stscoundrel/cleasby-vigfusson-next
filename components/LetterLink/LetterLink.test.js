import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import LetterLink from './index'

describe('LetterLink component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    ReactDOM.render(<LetterLink letter="a" />, div)
    ReactDOM.unmountComponentAtNode(div)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<LetterLink letter="s" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
