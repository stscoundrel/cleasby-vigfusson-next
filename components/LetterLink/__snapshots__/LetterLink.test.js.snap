// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LetterLink component Matches snapshot 1`] = `
<a
  href="/letter/s"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  s
</a>
`;
